.evergreen/generate_evergreen_tasks.js

```diff
@@ -336,8 +336,34 @@ for (const VERSION of AWS_AUTH_VERSIONS) {
     ]
   }));
 
-  TASKS.push(...aws_tasks);
-  AWS_AUTH_TASKS.push(...aws_tasks.map(t => t.name));
+  const awsNoOptionalTasks = awsFuncs.map(fn => ({
+    name: `${name(fn.func)}-no-optional`,
+    commands: [
+      {
+        func: 'install dependencies',
+        vars: {
+          NPM_OPTIONS: '--no-optional'
```

Comment:
We run the aws tests twice each now, once with the sdk installed and once without.